src/CaptureServiceBase/CaptureServiceBase.cpp

```diff
@@ -72,13 +72,21 @@ void CaptureServiceBase::FinalizeEventProcessing(StopCaptureReason stop_capture_
   ProducerCaptureEvent capture_finished;
   switch (stop_capture_reason) {
     case StopCaptureReason::kClientStop:
+    case StopCaptureReason::kGuestOrcStop:
       capture_finished = CreateSuccessfulCaptureFinishedEvent();
       break;
     case StopCaptureReason::kMemoryWatchdog:
-      capture_finished = CreateMemoryThresholdExceededCaptureFinishedEvent();
+      capture_finished =
+          CreateInterruptedByServiceCaptureFinishedEvent("OrbitService was using too much memory.");
       break;
     case StopCaptureReason::kExceededMaxDurationLimit:
-      capture_finished = CreateMaxCaptureDurationExceededCaptureFinishedEvent();
+      capture_finished = CreateInterruptedByServiceCaptureFinishedEvent(
+          "Capture duration exceeded the maximum duration limit.");
+      break;
+    case StopCaptureReason::kGuestOrcConnectionFailure:
+      capture_finished =
+          CreateFailedCaptureFinishedEvent("Encountered GuestOrc connection failure.");
```

Comment:
Purely for the sake of consistency with the other ones, I would make this a "full sentence" too (not sure what the right terminology is), e.g., "Connection with GuestOrc failed." or "A failure in the connection with GuestOrc was encountered."